pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -284,12 +275,13 @@ func (l *AccessCredentialManager) agentSetFilePermissions(domName string, filePa
 	return nil
 }
 
-func (l *AccessCredentialManager) agentSetUserPassword(domName string, user string, password string) error {
+func (l *AccessCredentialManager) agentSetUserPassword(domName, user, password string) (err error) {
 	domain, err := l.virConn.LookupDomainByName(domName)
 	if err != nil {
 		return fmt.Errorf("domain lookup failed: %w", err)
 	}
-	defer domain.Free()
+	defer func() { err = errors.Join(err, domain.Free()) }()
```

Comment:
Are you sure it won't be masked when returning `domain.SetUserPassword`?

Reply:
No, the return statement will assign value to `err` and then defer will be run, and another value will be assigned to `err`. It works as expected:
https://go.dev/play/p/eT-aVFlqMt_O?v=goprev